http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -249,18 +267,24 @@ private void updateResponseMetadata(EndpointResponse response, ActionPayload pay
   }
 
   private ActionPayload handleErrorResponse(ActionRequest request, String statusCode,
-      String statusMessage) {
-    return ActionPayload.error(request, statusCode, statusMessage);
-  }
-
-  private ActionPayload handleInvalidResponseBodyFormat(ActionRequest request, DecodeException e) {
-    return ActionPayload.error(request, "Response body is not a valid JSON!", e.getMessage());
+      String statusMessage, ActionLogger actionLogger) {
+    ActionPayload payload = ActionPayload.error(request, statusCode, statusMessage);
+    actionLogger.error(RESPONSE, payload.getResponse().toJson());
+    return payload;
   }
 
-  private ActionPayload handleSuccessResponse(EndpointResponse response, ActionRequest request) {
-    if (isForceJson || isJsonPredicate) {
-      return ActionPayload.success(request, bodyToJson(response.getBody().toString()));
-    } else if (isContentTypeHeaderJson(response)) {
+  private ActionPayload handleSuccessResponse(EndpointResponse response, ActionRequest request,
+      ActionLogger actionLogger) {
+    ActionPayload payload = ActionPayload
```

Comment:
Can you please check what data is logged in `ActionPayload.getResponse()`? I am afraid that there is only information about success / failure. The response message has the format:
```
Response      = Status-Line               ; Section 6.1
                       *(( general-header        ; Section 4.5
                        | response-header        ; Section 6.2
                        | entity-header ) CRLF)  ; Section 7.1
                       CRLF
                       [ message-body ]          ; Section 7.2
```
where `status-line` contains:
```
Status-Line = HTTP-Version SP Status-Code SP Reason-Phrase CRLF
```
So we can log:
- http version
- status code
- status message
- headers
- tailers

See `io.vertx.reactivex.ext.web.client.HttpResponse` for more details.

Additionally you can improve `logResponseData()` and reuse its logic.